br refactor (3): translate use cases, faq, increment br, monitor, batch create, and auto tune (#11379)
pingcap/docs: br/br-auto-tune.md

---
title: Backup Auto-Tune
summary: Learn about the auto-tune feature of TiDB backup and restore, which automatically limits the resources used by backups to reduce the impact on the cluster in case of high cluster resource usage.
---

# Backup Auto-Tune <span class="version-mark">New in v5.4.0</span>

> Before TiDB v5.4.0, when you back up data using br command-line tool, the number of threads used for backup makes up 75% of the logical CPU cores. Without a speed limit, the backup process can consume a lot of cluster resources, which has a considerable impact on the performance of the online cluster. Although you can reduce the impact of backup by adjusting the size of the thread pool, it is a tedious task to observe the CPU load and manually adjust the thread pool size.

Reviewer: ```suggestion
Before TiDB v5.4.0, when you back up data using BR, the number of threads used for backup makes up 75% of the logical CPU cores. Without a speed limit, the backup process can consume a lot of cluster resources, which has a considerable impact on the performance of the online cluster. Although you can reduce the impact of backup by adjusting the size of the thread pool, it is a tedious task to observe the CPU load and manually adjust the thread pool size.
```

> To reduce the impact of backup tasks on the cluster, starting from TiDB v5.4.0, br command-line tool introduces the auto-tune feature, which is enabled by default. When the cluster resource utilization is high, br command-line tool automatically limits the resources used by backup tasks and thereby reduces their impact on the cluster. The auto-tune feature is enabled by default.

Reviewer: ```suggestion
To reduce the impact of backup tasks on the cluster, starting from TiDB v5.4.0, BR introduces the auto-tune feature, which is enabled by default. When the cluster resource utilization is high, TiDB backup feature automatically limits the resources used by backup tasks and thereby reduces their impact on the cluster. The auto-tune feature is enabled by default.
```

## User scenario

> If you want to reduce the impact of backup tasks on the cluster, you can enable the auto-tune feature. With this feature enabled, br command-line tool performs backup tasks as fast as possible without excessively affecting the cluster.

Reviewer: ```suggestion
If you want to reduce the impact of backup tasks on the cluster, you can enable the auto-tune feature. With this feature enabled, TiDB performs backup tasks as fast as possible without excessively affecting the cluster.
```

The auto-tune feature has the following issues and corresponding solutions:

> - Issue 1: For **write-heavy clusters**, auto-tune might put the workload and backup tasks into a "positive feedback loop": the backup tasks take up too many resources, which causes the cluster to use fewer resources; at this point, auto-tune might mistakenly assume that the cluster is not under heavy workload and thus allowing br command-line tool to run faster. In such cases, auto-tune is ineffective.

Reviewer: ```suggestion
- Issue 1: For **write-heavy clusters**, auto-tune might put the workload and backup tasks into a "positive feedback loop": the backup tasks take up too many resources, which causes the cluster to use fewer resources; at this point, auto-tune might mistakenly assume that the cluster is not under heavy workload and thus allowing backup to run faster. In such cases, auto-tune is ineffective.
```

## Implementation

> Auto-tune adjusts the size of the thread pool used by backup tasks using the br tool to ensure that the overall CPU utilization of the cluster does not exceed a specific threshold.

Reviewer: ```suggestion
Auto-tune adjusts the size of the thread pool used by backup tasks to ensure that the overall CPU utilization of the cluster does not exceed a specific threshold.
```
